ethereumetl/deduplication/deduplicate.py

```diff
@@ -0,0 +1,67 @@
+import os
+import logging
+import asyncio
+from datetime import datetime
+from ethereumetl.constants import constants
+
+def deduplicate_records(records, ts_key, db):
+    ch_fallback_days = int(os.environ.get('CLICKHOUSE_FALLBACK_DAYS', constants.CLICKHOUSE_FALLBACK_DAYS))
+    
+    if records == None or len(records) == 0:
+        return records
+    
+    min_ts = get_minimum_ts(records, ts_key)
+    if is_ts_older(min_ts, ch_fallback_days):
+        records = asyncio.run(filter_records(records, min_ts, db))
+    return records
+
+def is_ts_older(ts, days):
+    difference = datetime.utcnow() - datetime.utcfromtimestamp(ts)
+    return difference.days > days    
+
+async def filter_records(items, min_ts_epoch, db):
+    if items == None or len(items) == 0:
+        return items
+    
+    message_type = items[0].get('type')
```

Comment:
All items will be of same type in the list?

Reply:
yes as we are running this deduplicate function for each type of message array